src/GitHub.UI/Helpers/SharedDictionaryManagerBase.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Windows;
+using System.Collections.Generic;
+
+namespace GitHub.Helpers
+{
+    public class SharedDictionaryManagerBase : ResourceDictionary
+    {
+        static IDictionary<Uri, ResourceDictionary> sharedDictionaries;
+        static IList<IDisposable> disposables;
+        Uri source;
+
+        static SharedDictionaryManagerBase()
+        {
+            // Avoid leaking resources at design time.
+            sharedDictionaries = GetAppDomainSharedDictionaries();
+            disposables = GetAppDomainDisposables();
+        }
+
+        public virtual new Uri Source
+        {
+            get { return source; }
```

Comment:
NullGuard will throw a NRE if the getter is called when `source` hasn't been set yet. Mark this with `[return: AllowNull]` to allow a null return, or make sure `source` is initialized to something to avoid the NRE.

(technically it won't be called before the setter, but the XamlDesigner might decide to do something wonky, and throwing here will cause it to go boom, so probably better safe than sorry)